alibi_detect/utils/random.py

```diff
@@ -0,0 +1,65 @@
+from contextlib import contextmanager
+import random
+import numpy as np
+import os
+from alibi_detect.utils.frameworks import has_pytorch, has_tensorflow
+
+if has_tensorflow:
+    import tensorflow as tf
+if has_pytorch:
+    import torch
+
+
+def set_seed(seed):
```

Comment:
`seed: int`
Is `None` valid/desirable here (e.g. you can do `np.random.seed(None)` do seed with an unknown state).

Reply:
Yes I wanted that, but annoyingly [torch.random.manual_seed](https://pytorch.org/docs/stable/generated/torch.manual_seed.html) doesn't support `None`. I guess we could take `None`, and then generate a random seed ourselves using system time etc, but I didn't want to complicate matters too much for now, especially since `random`, `np.random` and `tf.random` all randomly generate seeds in different ways if `None` is passed.

Reply:
On second thoughts, in the event of `None`, we could call `np.random.seed(None)` first, and then get that seed to use to set the others?

Reply:
I don't think it's worth complicating at this point, the utility of a "random random seed" is also not clear in a function specifically for setting a known seed.